### PR TITLE
build: changing sphinx-autoapi from 3.1.a2 to 3.1.a4

### PR DESCRIPTION
**Pull request by RobPasMue**
Dev branch to modify ``sphinx-autoapi`` version to expected 3.1.0 release...

@jorgepiloto - for visibility

### CODE REVIEW

**Comment by RobPasMue** [comment]
Trying with a4...

File: pyproject.toml [hunk 1]
```diff
@@ -85,7 +85,7 @@ doc = [
     "pyvista[jupyter]==0.43.3",
     "requests==2.31.0",
     "sphinx==7.2.6",
-    "sphinx-autoapi==3.1.a2",
+    "sphinx-autoapi==3.1.a3",
```
Comment (RobPasMue) [inline pyproject.toml]:
```suggestion
    "sphinx-autoapi==3.1.a4",
```

**Comment by RobPasMue** [comment]
Okay... so new version seems to work fine! Problem is now related to our ``ansys-sphinx-theme`` templates...

**Comment by RobPasMue** [comment]
We'll need to wait for https://github.com/ansys/ansys-sphinx-theme/pull/347 to be merged and released. Looking forward to it.

File: doc/source/conf.py [hunk 2]
```diff
@@ -194,7 +194,7 @@ def get_wheelhouse_assets_dictionary():
     "show-module-summary",
     "special-members",
 ]
-autoapi_template_dir = get_autoapi_templates_dir_relative_path(Path(__file__))
+# autoapi_template_dir = get_autoapi_templates_dir_relative_path(Path(__file__))
```
Comment (jorgepiloto) [inline doc/source/conf.py:197]:
```suggestion
autoapi_template_dir = get_autoapi_templates_dir_relative_path(Path(__file__))
```

**Review by jorgepiloto — APPROVED** [review]

File: doc/make.bat [hunk 3]
```diff
@@ -8,7 +8,7 @@ if "%SPHINXBUILD%" == "" (
 	set SPHINXBUILD=sphinx-build
 )
 if "%SPHINXOPTS%" == "" (
-	set SPHINXOPTS=-j auto -W --color
+	set SPHINXOPTS=-j auto --color
```
Comment (RobPasMue) [inline doc/make.bat:11]:
```suggestion
	set SPHINXOPTS=-j auto -W --color
```

**Review by jorgepiloto — APPROVED** [review]

**Pull request merged**